src/Microsoft.Health.Dicom.Blob/Registration/DicomServerBuilderBlobRegistrationExtensions.cs

```diff
@@ -17,59 +17,113 @@ namespace Microsoft.Extensions.DependencyInjection
+            serverBuilder
+                .AddStorage<BlobContainerConfigurationAware>(configuration)
+                .AddStorage<MetadataContainerConfigurationAware>(configuration);
+
+            return serverBuilder;
         }
 
-        private static IDicomServerBuilder AddBlobPersistence(this IDicomServerBuilder serverBuilder, IConfiguration configuration)
+        private static IDicomServerBuilder AddStorage<TStoreConfigurationAware>(
+            this IDicomServerBuilder serverBuilder, IConfiguration configuration)
+            where TStoreConfigurationAware : IStoreConfigurationAware, new()
         {
-            IServiceCollection services = serverBuilder.Services;
+            var config = new TStoreConfigurationAware();
+            var blobConfig = configuration.GetSection(BlobServiceClientOptions.DefaultSectionName);
 
-            IConfiguration blobConfig = configuration.GetSection(BlobServiceClientOptions.DefaultSectionName);
-            services
-                .AddBlobServiceClient(blobConfig)
+            serverBuilder.Services
                 .AddBlobContainerInitialization(x => blobConfig
                     .GetSection(BlobInitializerOptions.DefaultSectionName)
                     .Bind(x))
-                .ConfigureContainer(Constants.ContainerConfigurationName, x => configuration
-                    .GetSection(DicomServerBlobConfigurationSectionName)
+                .ConfigureContainer(config.Name, x => configuration
+                    .GetSection(config.SectionName)
                     .Bind(x));
 
+            return serverBuilder;
+        }
+
+        /// <summary>
+        /// Adds the metadata store for the DICOM functions.
+        /// </summary>
+        /// <param name="functionsBuilder">The DICOM functions builder instance.</param>
+        /// <param name="containerName">The name of the metadata container.</param>
+        /// <param name="configuration">The configuration for the function.</param>
+        /// <returns>The functions builder.</returns>
+        public static IDicomFunctionsBuilder AddMetadataStorageDataStore(
+            this IDicomFunctionsBuilder functionsBuilder,
+            IConfiguration configuration,
+            string containerName)
+        {
+            EnsureArg.IsNotNull(functionsBuilder, nameof(functionsBuilder));
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            var blobConfig = configuration.GetSection(BlobServiceClientOptions.DefaultSectionName);
+            functionsBuilder.Services
+                .AddBlobServiceClient(blobConfig)
+                .AddPersistence<IMetadataStore, BlobMetadataStore, LoggingMetadataStore>()
+                .Configure<BlobContainerConfiguration>(MetadataContainerConfigurationAware.ConfigurationSectionName, c => c.ContainerName = containerName);
+
+            return functionsBuilder;
+        }
+
+        private static IServiceCollection AddPersistence(this IServiceCollection services)
```

Comment:
Is there much benefit from having this method vs replacing the one call to it with 
```
.AddPersistence<IFileStore, BlobFileStore, LoggingFileStore>()
.AddPersistence<IMetadataStore, BlobMetadataStore, LoggingMetadataStore>()
```

Reply:
No other benefit other than increasing the lines of code in a single method, as we continue to add Workitem related changes. I can move it into the main method.

Reply:
Done.